synapse/storage/databases/main/events_worker.py

```diff
@@ -2589,6 +2600,37 @@ async def have_finished_sliding_sync_background_jobs(self) -> bool:
             )
         )
 
+    async def get_invite_count_by_user(self, user_id: str, from_ts: int) -> int:
+        """
+        Get the number of invites sent by the given user after the provided timestamp.
+
+        Args:
+            user_id: user ID to search against
+            from_ts: a timestamp in milliseconds from the unix epoch
+
+        """
+
+        def _get_invite_count_by_user_txn(
+            txn: LoggingTransaction, user_id: str, from_ts: int
+        ) -> int:
+            sql = """
+                  SELECT COUNT(e.event_id) FROM events e JOIN room_memberships rm ON e.event_id = rm.event_id WHERE e.sender = ? AND e.received_ts > ? AND e.state_key != ? AND rm.membership = 'invite'
+            """
```

Comment:
Some updates:

 - Wrap and format
 - `USING` statement since we're matching the same column
 - explicit `INNER JOIN` (no reason besides it made me question what kind of join it was initially)
 - Remove `AND e.state_key != ?` condition (see other discussion)
 - Needs a `type = 'm.room.member'` condition to use the new index

```suggestion
            sql = """
                  SELECT COUNT(e.event_id)
                  FROM events e
                  INNER JOIN room_memberships rm USING (event_id)
                  WHERE
                      e.sender = ? AND rm.membership = 'invite'
                      AND type = 'm.room.member' AND e.received_ts > ?
            """
```

This should be updated on the decision from https://github.com/element-hq/synapse/pull/17948#discussion_r1868086802 though